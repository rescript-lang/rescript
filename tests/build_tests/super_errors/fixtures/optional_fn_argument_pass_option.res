let optFn = (~x: option<int>=?) => x

let t = Some(1)

let f = optFn(~x=t)
