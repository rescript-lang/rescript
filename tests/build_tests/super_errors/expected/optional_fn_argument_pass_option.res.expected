
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/optional_fn_argument_pass_option.res[0m:[2m5:18[0m

  3 [2mâ”‚[0m let t = Some(1)
  4 [2mâ”‚[0m 
  [1;31m5[0m [2mâ”‚[0m let f = optFn(~x=[1;31mt[0m)
  6 [2mâ”‚[0m 

  This has type: [1;31moption<int>[0m
  But this optional function argument is expecting: [1;33mint[0m
  
  You're passing an optional value into an optional function argument.
  Optional function arguments expect you to pass the concrete value, not an option, when passed directly.
             
  Possible solutions: 
  - Unwrap the option and pass a concrete value directly
  - If you really do want to pass the optional value, prepend the value with [1;33m?[0m to show you want to pass the option, like: [1;33m?t[0m